Counter/src/app/shared/user-service.ts

```diff
@@ -0,0 +1,78 @@
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable, map, BehaviorSubject, of, catchError } from 'rxjs';
+import { User } from './user';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class UserService {
+    private isLoggedInStream: BehaviorSubject<boolean> = new BehaviorSubject(false);
+    public isLoggedIn: Observable<boolean>;
+    private idStream: BehaviorSubject<number> = new BehaviorSubject(0);
+    public id: Observable<number>;
+    private userUrl: string = 'api/user'
+    httpOptions = {
+        headers: new HttpHeaders({ 'Content-Type': 'application/json' })
+    };
+
+    constructor(private http: HttpClient) { 
+        this.id = this.idStream.asObservable();
+        this.isLoggedIn = this.isLoggedInStream.asObservable();
+    }
+
+    getUser(id: number) {
```

Comment:
You should declare what this function returns. Also make sure you're using a linter like eslint to make sure that your IDE is warning you about these things.

Reply:
I don't think I've been using a linter, I'll look into that!

Reply:
look into eslint